test/rebar_compile_SUITE.erl

```diff
@@ -1608,6 +1608,28 @@ override_del_deps(Config) ->
               {dep, "dep_d"}]}
     ).
 
+override_del_pkg_deps(Config) ->
+    Deps = rebar_test_utils:expand_deps(pkg, [{"some_dep", "0.0.1", [{"other_dep", "0.0.1", []}]}]),
+    TopDeps = rebar_test_utils:top_level_deps(Deps),
+
+    {_, PkgDeps} = rebar_test_utils:flat_deps(Deps),
+    mock_pkg_resource:mock([{pkgdeps, PkgDeps}]),
+
+    RebarConfig = [
+        {deps, TopDeps},
+        {overrides, [
+            {del, some_dep, [
+                {deps, [other_dep]}
```

Comment:
What is this inner part? Are you sure you need this? Shouldn't it just be `{del, some_dep}`?

Reply:
This allows to only download one entry in the deps tuple as defined in some_dep. Seemed right to me

Reply:
Oh, I did the test with the inner part to, as Fred said, override one the `other_dep` from `some_dep`, since in the issue the example you gave with `eredis_cluster` and `ex_doc` is the same.

Reply:
Oh, I completely misread this. It is kinda confusing to read...

So it is saying "delete dep 'other_dep' from 'some_dep's list of deps"?

Reply:
Yep. A real example would be the one you mentioned in the issue:

*The `eredis_cluster` hex package includes `ex_doc` which can't be built with rebar3 so it fails. The solution is to override all deps but really this should work too:*

```erlang
{overrides, [{del, eredis_cluster, [{deps, [ex_doc]}]}]}.
```

If you prefer the override in the test to be a one-liner too I can change that :)

Reply:
Thanks, no, this is good.